NumLib/DOF/MeshComponentMap.cpp

```diff
@@ -117,20 +117,64 @@ MeshComponentMap::MeshComponentMap(
 
 MeshComponentMap MeshComponentMap::getSubset(
     std::vector<int> const& component_ids,
-    MeshLib::MeshSubset const& mesh_subset) const
+    std::vector<MeshLib::MeshSubset> const& bulk_mesh_subsets,
+    MeshLib::MeshSubset const& new_mesh_subset) const
 {
+    // Mapping of the nodes in the new_mesh_subset to the bulk mesh nodes
+    auto const& new_mesh_properties = new_mesh_subset.getMesh().getProperties();
+    if (!new_mesh_properties.template existsPropertyVector<std::size_t>(
+            "bulk_node_ids"))
+    {
+        OGS_FATAL(
+            "Bulk node ids map expected in the construction of the mesh "
+            "subset.");
+    }
+    auto const& bulk_node_ids_map =
+        *new_mesh_properties.template getPropertyVector<std::size_t>(
+            "bulk_node_ids");
+
     // New dictionary for the subset.
     ComponentGlobalIndexDict subset_dict;
 
-    std::size_t const mesh_id = mesh_subset.getMeshID();
+    std::size_t const new_mesh_id = new_mesh_subset.getMeshID();
     // Lookup the locations in the current mesh component map and
-    // insert the full lines into the subset dictionary.
-    for (std::size_t j = 0; j < mesh_subset.getNumberOfNodes(); j++)
+    // insert the full lines into the new subset dictionary.
+    for (auto* const node : new_mesh_subset.getNodes())
+    {
+        auto const node_id = node->getID();
+
+        MeshLib::Location const new_location{
+            new_mesh_id, MeshLib::MeshItemType::Node, node_id};
+
+        // Assuming the meshes for the all bulk_mesh_subsets are equal.
+#ifndef NDEBUG
+        {  // Testing the assumption.
```

Comment:
The test does not depend on the node, and could be tested before the loop and not only in debug mode.